### ISSUE TITLE
Multiple Users Problem

### ISSUE THREAD

**Issue by vinsilsim** [AHAAAAAAA/PokemonGo-Map#2123]
Hello Guys,

I have problem when multiple users use the map. When someone selects a new point, he stops the old point without finish and a start new point. I do not know if I was clear. I need a distro for 1000+ people simultaneously.

Possible?

**Comment by devinjparsons** [comment]
The old code used to finish its search before scanning new location, but the use case for almost all other users was they wanted new location scanned right away.  You or someone would have to change a couple lines that check for a new location in the middle of the scan to make it wait till the end of the scan.

**Comment by neruve** [comment]
Correct me if I'm wrong but I don't think the open source nature of this project is designed to have 1000+ users simultaneously use it from one host. Not only that but even with a fantastic home server, I doubt you would complete 1000+ different scans before pokemon started to expire.

**Comment by devinjparsons** [comment]
@neruve exactly, this project was meant for people to deploy on their own instead of relying on one of the major publicly available map.  @vinsilsim if you are wanting to implement this for that many users, you will need to make major changes to the app to handle what you are wanting.

**Comment by jakeman2048** [comment]
@vinsilsim or you could just run multiple instances with the '-ns' argument, but you would still have to set the fixed location and wouldn't be changeable via the web UI.

**Comment by callmebackdraft** [comment]
As @jakeman2048 stated multiple instances could solve alot,
I am currently running 7 instances constantly, of wich 6 are run with the -ns argument in the command line. This means none of those locations can be changed by users.

The other one is running the webserver, this means its location is variable and changable by moving the marker or typing a location in the options menu

**Comment by Soulofdestiny** [comment]
@callmebackdraft but the -ns instances are not shown in the webui as well, aren't they? If yes, can you explain shortly how to achieve this?

**Comment by GareRam** [comment]
@Soulofdestiny First you start your first instance, without the -ns argument. optionally you define -H x.x.x.x for the ip you want to bind your server to, along with -P xxxx for the port.

let's say you start with -H 10.0.0.10 -P 5050

This binds your server to that ip and port 5050. Now you start a second worker with the -ns argument. This worker scans to the same database. If you place the center of the working area next to the area your webserving worker is scanning, they will together scan to the same database, resulting in everything showing in the same map, on port 5050.

This also works for defining different area's to different ports. Lets say you've got a worker in city A, and a worker in city B, you can define different ports for them. You can however, still view city A with the city B map, and vice versa.

TL;DR: by default everything is scanned to the same database.

**Comment by Soulofdestiny** [comment]
Thanks - got it :). Now let's see how many processes my raspberry can handle ;)

**Comment by BenVlodgi** [comment]
Can you start up a worker that doesn't bother with the webui at all, and just focuses on scanning?

**Comment by jakeman2048** [comment]
Yeah, -ns does this. -os runs only webui.
